### PR TITLE
Reader: Fix how we detect Discover posts

### PR DESCRIPTION
**Pull request by blowery**
Prevents the Block option from appearing on Discover posts that don't have discover_metadata.

### CODE REVIEW

**Comment by blowery** [comment]
To test, pull up Discover in the Reader and expand the ellipsis menu for each post. None should include Block as an option. 

Make sure you're unproxied while testing and not an editor on discover.wordpress.com.

cc @kjellr

File: client/reader/discover/helper.jsx [hunk 1]
```diff
@@ -17,11 +18,11 @@ module.exports = {
 	},
 
 	isDiscoverPost: function( post ) {
-		return !! post.discover_metadata;
+		return !! ( post.discover_metadata || post.site_ID == config( 'discover_blog_id' ) ); //eslint-disable-line eqeqeq
```
Comment (TooTallNate) [inline client/reader/discover/helper.jsx]: What's the thought for not using `===` here?

Comment (blowery) [inline client/reader/discover/helper.jsx]: site_ID is sometimes a Number, sometimes a string. Depends on where we get it from.

Comment (blowery) [inline client/reader/discover/helper.jsx]: actually i might be wrong about that. it may always be a number.

Comment (blowery) [inline client/reader/discover/helper.jsx]: switched to === in https://github.com/Automattic/wp-calypso/pull/4833/commits/28c011be17049738c6413d1673ab9752ad271d50

**Pull request merged**